FutureNHS/futurenhs-platform
# Argocd manages branch clusters.

<img width="238" alt="Screenshot 2020-07-22 at 13 02 56" src="https://user-images.githubusercontent.com/13286587/88174157-b620af00-cc1b-11ea-8338-8ebed7b7f92f.png">

This PR primarily relates to setting up ArgoCD to manage dev clusters, and lays the groundwork to install Azure Service Operator and the PostgreSQL database which it will manage.

- [x] Update branch workflows to include clone / copy manifests/ deploy steps.
- [x] Add 'out of retries' message to master workflows.
- [x] Tweak 'Working with Sealed Secrets.md' for better readability. 
- [x] Fix secretKeyRef and update related Sealed Secret 
- [x] Update infrastucture README.
- [x] Add branch to dev-overlay variables 

## Testing
1. Update BRANCH variable in infrastructure/dev-overlay-variables.json to match this branch name.
2. Run ./install-argo script to update your local argo apps.
3. Run kubectl port-forward svc/argocd-server -n argocd 8080:443 to launch UI at localhost:8080.
4. Login using admin as user and set the passord to the name of the argocd server pod (kubectl get pods -n argocd to find it)
4. Verify in UI
